### PR TITLE
Add CalicoNodeStatus resource

### PR DESCRIPTION
**Pull request by song-jiang**
## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -39,4 +39,7 @@ require (
 	sigs.k8s.io/kind v0.11.1
 )
 
-replace github.com/sirupsen/logrus => github.com/projectcalico/logrus v1.0.4-calico
+replace (
+	github.com/projectcalico/api => github.com/song-jiang/api v0.1.2-0.20210913143137-f981580f3483
```
Comment (coutinhop) [inline go.mod:43]: do we really want to use the fork here?

Comment (song-jiang) [reply go.mod]: No. I would update the api pin once my api PR get merged.

File: lib/backend/k8s/resources/caliconodestatus.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	CalicoNodeStatusCRDName      = "CalicoNodeStatuses.crd.projectcalico.org"
+)
+
+func NewCalicoNodeStatusClient(c *kubernetes.Clientset, r *rest.RESTClient) K8sResourceClient {
+	return &customK8sResourceClient{
+		clientSet:       c,
+		restClient:      r,
+		name:            CalicoNodeStatusCRDName,
+		resource:        CalicoNodeStatusResourceName,
+		description:     "Calico CalicoNodeStatus",
```
Comment (caseydavenport) [inline lib/backend/k8s/resources/caliconodestatus.go:38]: Probably don't need to say "Calico CalicoNodeStatus", that's just too many Calicos!

File: lib/clientv3/caliconodestatus.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+// calicoNodeStatus implements CalicoNodeStatusInterface
+type calicoNodeStatus struct {
+	client client
+}
+
+// Create takes the representation of a CalicoNodeStatus and creates it.  Returns the stored
+// representation of the CalicoNodeStatus, and an error, if there is any.
+func (r calicoNodeStatus) Create(ctx context.Context, res *apiv3.CalicoNodeStatus, opts options.SetOptions) (*apiv3.CalicoNodeStatus, error) {
+	if err := validator.Validate(res); err != nil {
```
[the remark this answers is not in the document]
Comment (song-jiang) [reply lib/clientv3/caliconodestatus.go:44]: Agree. I'll add it later.

[the remark this answers is not in the document]
Comment (song-jiang) [reply lib/clientv3/caliconodestatus.go:44]: I added validations
- classes : validate:"required,unique"`
- UpdatePeriodSeconds: validate:"required,gte=0,lte=86,400"

**Review by caseydavenport — APPROVED** [review]

**Pull request merged**